### PR TITLE
Adding InteractiveBrowserCredential to Azure.Identity

### LINKED ISSUE #6928: Identity: Implement Interactive Authentication Support
Gets the token through an interactive process that prompts the user for credentials through a browser or pop-up window

### PR DESCRIPTION
**Pull request by schaabs**
closes #6928

### CODE REVIEW

File: sdk/identity/Azure.Identity/src/InteractiveBrowserCredential.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        /// <param name="options">The client options for the newly created DeviceCodeCredential</param>
+        public InteractiveBrowserCredential(string clientId, IdentityClientOptions options)
+        {
+            _clientId = clientId ?? throw new ArgumentNullException(nameof(clientId));
+
+            _options = options ??= new IdentityClientOptions();
+
+            var pipeline = HttpPipelineBuilder.Build(_options, bufferResponse: true);
+
+            _pubApp = PublicClientApplicationBuilder.Create(_clientId).WithHttpClientFactory(new HttpPipelineClientFactory(pipeline)).WithRedirectUri("http://localhost").Build();
```
Comment (pakrym) [inline sdk/identity/Azure.Identity/src/InteractiveBrowserCredential.cs]: Would we have problems with trying to bind to port 80? It's often taken.

File: sdk/identity/Azure.Identity/src/InteractiveBrowserCredential.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        }
+
+        /// <summary>
+        /// Obtains an <see cref="AccessToken"/> token for a user account silently if the user has already authenticated, otherwise the default browser is launched to authenticate the user.
+        /// </summary>
+        /// <param name="scopes">The list of scopes for which the token will have access.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>An <see cref="AccessToken"/> which can be used to authenticate service client calls.</returns>
+        public override AccessToken GetToken(string[] scopes, CancellationToken cancellationToken = default)
+        {
```
Comment (pakrym) [inline sdk/identity/Azure.Identity/src/InteractiveBrowserCredential.cs]: At this point just call GetTokenAsync().GetAwaiter().GetResult(), I don't think that having two copies of the method gives you anything.

File: sdk/identity/Azure.Identity/src/InteractiveBrowserCredential.cs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        /// <param name="clientId">The client id of the application to which the users will authenticate.</param>
+        /// TODO: need to link to info on how the application has to be created to authenticate users, for multiple applications
+        public InteractiveBrowserCredential(string clientId)
+            : this (clientId, null)
+        {
+
+        }
+
+        /// <summary>
+        /// Creates a new DeviceCodeCredential with the specifeid options, which will authenticate users with the specified application.
```
Comment (chlowell) [inline sdk/identity/Azure.Identity/src/InteractiveBrowserCredential.cs]: Docs copied from `DeviceCodeCredential`?

**Review by jianghaolu — APPROVED** [review]

File: sdk/identity/Azure.Identity/src/InteractiveBrowserCredential.cs (hunk 1 above)
Comment (jianghaolu) [reply sdk/identity/Azure.Identity/src/InteractiveBrowserCredential.cs]: Seems like it's required by MSAL: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/MSAL.NET-uses-web-browser. But I'm also curious about identity team's decision behind it.

Comment (jianghaolu) [reply sdk/identity/Azure.Identity/src/InteractiveBrowserCredential.cs]: Also on *nix machines you'll need root to open ports < 1024.

Comment (schaabs) [reply sdk/identity/Azure.Identity/src/InteractiveBrowserCredential.cs]: Created https://github.com/Azure/azure-sdk-for-net/issues/7086 to address this.

**Pull request merged**